### PR TITLE
Create an FX Module for Archival

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
- I added a Module to the service/history/archival package and embedded it in the history module
- I also modified our existing tests to use this module instead of the raw constructor to improve test coverage

**Why?**
I made this change so that we can inject the required dependencies into the archival queue

**How did you test it?**
I tested this by modifying our tests to use this module instead of the normal constructor, and I then checked test coverage as well. I also ran a simple `make start`. Additionally, this dependency isn't required anywhere yet, so it shouldn't actually be run in prod.

**Potential risks**
None.

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/workflow/fx.go [hunk 1]
```diff
@@ -30,4 +30,5 @@ import (
 
 var Module = fx.Options(
 	fx.Populate(&taskGeneratorProvider),
+	fx.Provide(NewRelocatableAttributesFetcher),
```
Comment (yux0) [inline service/history/workflow/fx.go:33]: nit: RelocatableAttributesFetcherProvider?

File: common/dynamicconfig/constants.go [hunk 2]
```diff
@@ -522,6 +522,8 @@ const (
 	ArchivalProcessorArchiveDelay = "history.archivalProcessorArchiveDelay"
 	// ArchivalProcessorRetryWarningLimit is the number of times an archival task may be retried before we log a warning
 	ArchivalProcessorRetryWarningLimit = "history.archivalProcessorRetryLimitWarning"
+	// ArchivalBackendMaxRPS is the maximum rate of requests per second to the archival backend
```
Comment (yux0) [inline common/dynamicconfig/constants.go:525]: where is the rate limiter used?

Comment (MichaelSnowden) [reply common/dynamicconfig/constants.go:525]: It's instantiated here: 
image

And then it's used here: 
image

File: service/history/workflow/fx.go (hunk 1 above)
Comment (MichaelSnowden) [reply service/history/workflow/fx.go]: Fixed

File: service/history/archival/fx.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	"go.temporal.io/server/common/quotas"
+	"go.temporal.io/server/service/history/configs"
+)
+
+var Module = fx.Options(
+	fx.Provide(NewArchiver),
+	fx.Provide(func(config *configs.Config) quotas.RateLimiter {
+		return quotas.NewRateLimiter(
+			config.ArchivalBackendMaxRPS(),
+			int(math.Ceil(config.ArchivalBackendMaxRPS())),
```
Comment (yycptt) [inline service/history/archival/fx.go:41]: This means changing rps requires a service restart to take effect, maybe use dynamic rate limiter?

Comment (MichaelSnowden) [reply service/history/archival/fx.go]: Oh, much better, thanks! Should I mark this old interface as deprecated?

**Review by yux0 — APPROVED** [review]

**Pull request merged**